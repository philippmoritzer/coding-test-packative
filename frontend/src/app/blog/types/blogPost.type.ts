export type BlogPost = {
  id: string;
  title: string;
  content: string;
  createdBy: string;
  likes: number;
  createdAt: string;
  updatedAt: string;
};

export type BlogPostCardProps = {
  blogPost: BlogPost;
};
