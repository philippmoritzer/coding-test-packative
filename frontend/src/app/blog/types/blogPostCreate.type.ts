export type BlogPostCreateType = {
  title: string;
  content: string;
  createdBy: string;
};
