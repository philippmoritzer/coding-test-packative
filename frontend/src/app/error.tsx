'use client';

import React from "react";
import Link from "next/link";

export default function Error({ error, reset }: { error: Error; reset: () => void }) {
  return (
    <div className="min-h-screen flex flex-col items-center justify-center bg-gradient-to-r from-gray-100 to-gray-200">
      <div className="bg-white shadow-lg rounded-lg p-10 max-w-lg w-full text-center">
        <h1 className="text-4xl font-bold text-red-600 mb-4">Something went wrong</h1>
        <p className="text-gray-700 mb-6">
          {error?.message || "An unexpected error has occurred. Please try again later."}
        </p>
        <div className="flex flex-col sm:flex-row gap-4 justify-center">
          <button
            onClick={reset}
            className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-2 rounded transition"
          >
            Try Again
          </button>
          <Link
            href="/blog"
            className="bg-gray-200 hover:bg-gray-300 text-gray-800 px-6 py-2 rounded transition"
          >
            Back to Blog
          </Link>
        </div>
      </div>
    </div>
  );
}