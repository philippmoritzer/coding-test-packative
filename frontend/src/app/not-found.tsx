'use client';

import React from "react";
import Link from "next/link";

export default function NotFound() {
  return (
    <div className="min-h-screen flex flex-col items-center justify-center bg-gradient-to-r from-gray-100 to-gray-200">
      <div className="bg-white shadow-lg rounded-lg p-10 max-w-lg w-full text-center">
        <h1 className="text-5xl font-bold text-blue-600 mb-4">404</h1>
        <h2 className="text-2xl font-semibold text-gray-800 mb-2">Page Not Found</h2>
        <p className="text-gray-700 mb-6">
          Sorry, the page you are looking for does not exist or has been moved.
        </p>
        <Link
          href="/blog"
          className="bg-blue-600 hover:bg-blue-700 text-white px-6 py-2 rounded transition"
        >
          Back to Blog
        </Link>
      </div>
    </div>
  );
}