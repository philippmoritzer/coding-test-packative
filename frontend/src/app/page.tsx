'use client'

import Link from "next/link";

function Main() {
  const haneldLogout = () => {
    localStorage.removeItem("authString");
    window.location.href = "/login";
  };
  return (
    <div >
      <Link href="/todo">Go to do page</Link>      
      <button className="bg-blue-500" onClick={haneldLogout}>Logout</button>

    </div>
  );
}

export default Main;
