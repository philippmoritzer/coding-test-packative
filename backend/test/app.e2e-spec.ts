import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { App } from 'supertest/types';
import { AppModule } from './../src/app.module';

describe('AppController (e2e)', () => {
  let app: INestApplication<App>;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    await app.init();
  });

  it('/ (GET) todo items', async () => {
    const res = await request(app.getHttpServer())
      .get('/todo/list')
      .set('Authorization', 'authenticated;permissions: get-todo')
      .expect(200);
    console.log(res.body);
    expect(res.body).toMatchSnapshot();
  });
});
