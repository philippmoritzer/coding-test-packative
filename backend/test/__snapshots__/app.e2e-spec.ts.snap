// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) / (GET) todo items 1`] = `
{
  "todoItems": [
    {
      "id": "87dad0b1-cbed-4f3a-ad7e-49cef5789166",
      "name": "Todo 1 description",
      "type": "EMPTY_DISHWASHER",
    },
    {
      "id": "791bb271-0d35-4a03-8c3c-d87a462a2f47",
      "name": "Todo 1 description",
      "type": "EMPTY_DISHWASHER",
    },
  ],
}
`;
